examples/codec_usage.py

```diff
@@ -0,0 +1,121 @@
+"""
+Example demonstrating custom codec usage with COMPAS EVE.
+
+This example shows how to:
+1. Use the default JsonMessageCodec
+2. Create a custom codec
+3. Use ProtobufMessageCodec (if compas_pb is installed)
+"""
+
+import compas_eve as eve
+from compas_eve import JsonMessageCodec, MessageCodec
+
+
+# Example 1: Using the default JSON codec (implicit)
+print("=" * 60)
+print("Example 1: Default JSON Codec (implicit)")
+print("=" * 60)
+
+pub = eve.Publisher("/example/default")
+sub = eve.EchoSubscriber("/example/default")
+sub.subscribe()
+
+pub.publish(eve.Message(text="Hello with default JSON codec", count=1))
+print()
+
+
+# Example 2: Explicitly using JSON codec
+print("=" * 60)
+print("Example 2: Explicit JSON Codec")
+print("=" * 60)
+
+json_codec = JsonMessageCodec()
+transport = eve.InMemoryTransport(codec=json_codec)
+
+pub = eve.Publisher("/example/json", transport=transport)
+sub = eve.EchoSubscriber("/example/json", transport=transport)
+sub.subscribe()
+
+pub.publish(eve.Message(text="Hello with explicit JSON codec", count=2))
+print()
+
+
+# Example 3: Custom codec
+print("=" * 60)
+print("Example 3: Custom Codec")
+print("=" * 60)
+
+
+class UpperCaseCodec(MessageCodec):
+    """A simple custom codec that converts all string values to uppercase."""
+
+    def encode(self, message):
+        """Encode message by converting all string values to uppercase."""
+        import json
+        # Extract data from message
+        try:
+            data = message.data
+        except (KeyError, AttributeError):
+            try:
+                data = message.__data__
+            except (KeyError, AttributeError):
+                data = dict(message)
```

Comment:
for the sake of example, there's no need to do the whole thing, assume `message` to be of type `Message` and that's it